### PR TITLE
AWS federation for GitLab Using OIDC

### PR DESCRIPTION
**Pull request by thaunghtike-share**
AWS federation for GitLab Using OIDC

### CODE REVIEW

File: versions.tf [hunk 1]
```diff
@@ -1,24 +0,0 @@
-// Copyright © 2021 Daniel Morris
```
Comment (abhiyerra) [inline versions.tf]: Why was this file removed?

File: LICENSE.md [hunk 2]
```diff
@@ -1,166 +0,0 @@
-# Apache License, Version 2.0
```
Comment (abhiyerra) [inline LICENSE.md]: Please don't delete this file.

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -1,52 +0,0 @@
...
-    branches:
-    - main
-    paths-ignore:
-    - .editorconfig
-    - .gitignore
-    - LICENSE.md
-    - README.md
-
-jobs:
-  verify:
```
Comment (abhiyerra) [inline .github/workflows/ci.yaml]: We want this part of the code.

File: README.md [hunk 4]
```diff
@@ -1,124 +1,181 @@
...
-```terraform
+```bash
 provider "aws" {
-  region = var.region
+  region = var.aws_region
 }
 
+provider "tls" {}
+
 module "aws_oidc_github" {
```
Comment (abhiyerra) [inline README.md:22]: This was not changed.

File: README.md [hunk 5]
```diff
@@ -1,124 +1,181 @@
...
 
-* [Configuring OpenID Connect in Amazon Web Services]
-* [Creating OpenID Connect (OIDC) identity providers]
-* [Obtaining the thumbprint for an OpenID Connect Identity Provider]
+- [Configure OpenID Connect in AWS to retrieve temporary credentials
+](https://docs.gitlab.com/ee/ci/cloud_services/aws/)
+- [Connect to cloud services
+](https://docs.gitlab.com/ee/ci/cloud_services/index.html#configure-a-conditional-role-with-oidc-claims)
 
-## License
```
Comment (abhiyerra) [inline README.md:113]: Please restore this section.

**Comment by abhiyerra** [comment]
Tasks before merging:

 - [ ] Run `terraform fmt -recursive`
 - [ ] Restore the LICENSE.md and add opsZero, LLC to the license.

File: examples/main.tf [hunk 6]
```diff
@@ -0,0 +1,12 @@
+module "aws_oidc_gitlab" {
```
Comment (abhiyerra) [inline examples/main.tf]: Move these files into the `complete` directory like the github oidc.

File: examples/providers.tf [hunk 7]
```diff
@@ -0,0 +1,22 @@
+terraform {
```
Comment (abhiyerra) [inline examples/providers.tf]: This section should be the versions.tf that you removed below.

File: examples/providers.tf [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    tls = {
+      source  = "hashicorp/tls"
+      version = "3.3.0"
+    }
+  }
+
+  required_version = "~> 1.0"
+}
+
+provider "aws" {
```
Comment (abhiyerra) [inline examples/providers.tf]: The providers can be in the main.tf and this file can be removed.

File: variables.tf [hunk 9]
```diff
@@ -66,31 +51,27 @@ variable "iam_role_path" {
...
 }
 
-variable "tags" {
-  default     = {}
-  description = "Map of tags to be applied to all resources."
-  type        = map(string)
+variable "match_field" {
+  type        = string
+  default     = "aud"
+  description = "	Issuer, the domain of your GitLab instance. Default value is 'aud' with gilab instance url. Change to 'sub' if you want to use the filter to any project"
```
Comment (abhiyerra) [inline variables.tf]: Can you remove the space in the description?

**Review by abhiyerra — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 9 above)
Comment (thaunggye) [reply variables.tf]: Ok I will fix them all now

File: LICENSE.md (hunk 2 above)
Comment (thaunghtike-share) [reply LICENSE.md]: Already added License

**Comment by thaunghtike-share** [comment]
I've fixed the issues. Could you pls check it ? but I don't know what this means - add opsZero, LLC to the license.

**Review by abhiyerra — APPROVED** [review]
